### ISSUE TITLE
Is it a good pattern to use ev-* properties?

### ISSUE THREAD

**Issue by bitinn** [Matt-Esch/virtual-dom#273]
I have some questions regarding `virtual-dom` best practice on `ev-*` properties:

- The more I use it and the more it feels like an anti-pattern to separation of concerns, why would you want your template (vdom component in this case) to have knowledge of your event handlers? So that you can hook/unhook them automagically within `virtual-dom` loop?

- If you want to re-use the same vdom component on 2 different page, they might not have the same event handler logic, but component itself shouldn't care about who's using them right? Having event handlers declared on them seem like a maintenance problem in long-run?

- Wouldn't a general dom delegator (like [dom-delegate](https://github.com/ftlabs/ftdomdelegate)) using data-* attributes and separate event store be a lot more flexible? What does `ev-*` do in particular that saves us from writing boilerplate code?

**TL;DR**: AFAIK, both `dom-delegator` and `mercury` are using `ev-*` to attach custom event emitter, and nothing more. So no business logic in the component itself, is it the way to go?

**Comment by neonstalwart** [comment]
> Wouldn't a general dom delegator (like dom-delegate) using data-* attributes and separate event store be a lot more flexible?

this is more or less what ev-* is doing already.  so i think you're suggesting the same thing except that ev-* doesn't use your data-* attributes, it uses another mechanism.

in reusable/generic components, you don't need to provide your business logic in the ev-* handlers but instead those could be triggers for "events" that consumer code uses to initiate business logic.  e.g. a button could have a generic "click" event and code which constructs the button component could add listeners that provide the business logic of what should happen when that button is clicked.

**Comment by bitinn** [comment]
I've been using the `ev-click: emitter.emit('event-name', data)` approach for a while, and it works well. Though I find myself needing to write a helper like this to avoid writing preventDefault everywhere with `dom-delegator`.

```
emitter.capture = function(name, data) {
	var self = this;
	return function(ev) {
		ev.preventDefault();
		self.emit(name, data);
	};
};
```

Just thought to share my experience before closing  :)
